static/js/tools/map/time_slider.tsx

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Time slider component.
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+
+interface TimeSliderProps {
+  // Current date of map
+  currentDate: string;
+
+  // Selected dates to display on time slider
+  dates: Array<string>;
+
+  // Hash representing current time series
+  metahash: string;
+
+  // Whether the slider is enabled on refresh
+  // False if the map is displaying multiple dates (e.g. 'Best Available')
+  startEnabled: boolean;
+
+  // Fetches data for slider dates when play is pressed
+  onPlay(metahash: string, callback: () => void): any;
```

Comment:
usually we want to avoid the `any` datatype. I think here, you can do `onPlay: (metahash:string, callback: () => void) => void` instead.

similar thing for `updateDate` and in the Chart component props as well